src/Aws/Common/Credentials/RefreshableInstanceProfileCredentials.php

```diff
@@ -42,6 +42,21 @@ public function __construct(CredentialsInterface $credentials, InstanceMetadataC
         $this->client = $client ?: InstanceMetadataClient::factory();
     }
 
+    public function serialize()
+    {
+        return json_encode(array(
+            'credentials' => parent::serialize(),
+            'client' => serialize($this->client),
```

Comment:
It can. There's no custom serialization defined, so `serialize` will just capture all object properties. The output is a bit large so I could cut out whatever is unnecessary.